open-api/rest-catalog-open-api.yaml

```diff
@@ -3647,6 +3826,151 @@ components:
             type: integer
           description: "List of equality field IDs"
 
+    DeleteFile:
+      oneOf:
+        - $ref: '#/components/schemas/PositionDeleteFile'
+        - $ref: '#/components/schemas/EqualityDeleteFile'
+
+    PreplanTableRequest:
+      type: object
+      oneOf:
+        - required:
+            - snapshot-id
+        - required:
+            - start-snapshot-id
```

Comment:
This is certainly closer but I think we'll need to make sure this is how we specify mutually exclusive fields. https://stackoverflow.com/questions/21134029/how-to-define-mutually-exclusive-query-parameters-in-swagger-openapi/49199240#49199240 

I know the link above is for URL query params but maybe the same principle applies? 

Also at minimum I think we need to document that servers must throw a 400 if neither a snapshot ID nor a start and end snapshot ID pair are specified.

Reply:
https://stackoverflow.com/questions/72218716/how-to-define-in-swagger-spec-either-of-two-properties-are-not-null-and-required/72247466#72247466 maybe a better example, but istead of `anyOf` it's `oneOf`

Reply:
@amogh-jahagirdar @rdblue 
Is this the cleanest way of expressing mutually exclusive fields? I feel like the examples mentioned usage of the `minProperties` and `minItems` kinda make the spec harder to read here.

Will try to do some more research on this, and see if can find anything simpler. If not can try to conform around this.

Reply:
I think a clean way of expressing mutually exclusive fields is to use the context structure that Jack introduced. Then just use `allOf` to incorporate the context into the request rather than adding it as a `scan-context` field.

Alternatively, we could make them optional in OpenAPI and document that you must send either `snapshot-id` or `start-snapshot-id` and `end-snapshot-id`. I don't have a problem with that since there isn't a very simple way to do this.

Reply:
@rdblue I think we removed the context since we didn't want to introduce more nesting, and just have the properties in the request objects themselves.

I can try doing the second approach you mentioned, as it seems straight forward to me.